nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/StreamWorkManagerTest.java

```diff
@@ -89,54 +94,56 @@ public void testWorkManagerConfigDisableAllAfterStart() throws Exception {
     @Test
     public void testWorkIdempotent() throws InterruptedException {
         MetricsTracker tracker = new MetricsTracker();
-        SleepWork work = new SleepWork(1000, false);
+        SleepWork work = new SleepWork(getDurationMillis());
         assertTrue(work.isIdempotent());
         service.schedule(work);
-        assertTrue(service.awaitCompletion(5, TimeUnit.SECONDS));
+        assertTrue(service.awaitCompletion(getDurationMillis() * 2, TimeUnit.MILLISECONDS));
         tracker.assertDiff(0, 0, 1, 0);
 
-        // schedule again the exact same work 5 times
-        service.schedule(work);
-        service.schedule(work);
+        // schedule again the exact same work 3 times
         service.schedule(work);
         service.schedule(work);
         service.schedule(work);
 
         // works with the same id are skipped immediately and marked as completed, we don't have to wait 5s
-        assertTrue(service.awaitCompletion(500, TimeUnit.MILLISECONDS));
-        tracker.assertDiff(0, 0, 6, 0);
+        assertTrue(service.awaitCompletion(getDurationMillis() / 2, TimeUnit.MILLISECONDS));
+        tracker.assertDiff(0, 0, 4, 0);
     }
 
+    /**
+     * This test cannot be run with storeState enabled.<br>
+     * When the first work finishes, its status is not scheduled anymore and following works are not run.
+     */
     @Test
+    @Deploy("org.nuxeo.ecm.core.event:test-stream-workmanager-disable-storestate.xml")
     public void testWorkNonIdempotent() throws InterruptedException {
         MetricsTracker tracker = new MetricsTracker();
-        SleepWork work = new SleepWork(1000, false);
+        int duration = getDurationMillis() * 2;
+        SleepWork work = new SleepWork(duration);
         work.setIdempotent(false);
         assertFalse(work.isIdempotent());
         service.schedule(work);
-        assertTrue(service.awaitCompletion(5, TimeUnit.SECONDS));
+        assertTrue(service.awaitCompletion(duration + 100, TimeUnit.MILLISECONDS));
```

Comment:
100ms of margin is too low, a CI slave can run dozen of jobs in concurrence and pause for seconds.
you can use long timeout here, the timeout will be reached only on error case.